src/containers/Map/MapScreen.tsx

```diff
@@ -213,6 +274,7 @@ const MapScreen = () => {
 	}
 
 	if (showMapNodes && selectedBotForOrder) {
+		let a: boolean = rerender;
```

Comment:
What exactly is the `rerender` state being used for? This is the only instance it is used, as far as I can tell.

Reply:
Basically when I add a route to the current route array the app won't rerender and display the changes. So it was kinda a hacky fix I added to change this state so the page would reload and the changes would display. Idk if you have a better idea

Reply:
I bet that's exactly because of the use of array mutating methods I mentioned. If you change the array methods you use to `concat` etc., React should detect the array has changed and rerender automatically.

Reply:
you were right. I changed it and took out the rerender. Thanks for letting me know!